src/Avalonia.Controls/TopLevel.cs

```diff
@@ -232,6 +235,18 @@ protected bool AutoSizing
         }
 
         /// <inheritdoc/>
+        IRenderTarget IRenderRoot.CreateRenderTarget()
+        {
+            return _renderInterface.CreateRenderer(PlatformImpl.Handle);
```

Comment:
I think we might even make `ITopLevelImpl` to actually implement `IRenderTarget` by itself. There are some cases (e. g. GTK embedding) when we are trying to obtain render target when native window haven't been created yet. I think we should ask for `DrawingContext` from `ITopLevelImpl` directly and allow it to return `null` if `DrawingContext` can't be created and skip render pass in this case. It will also allow to skip frames when window isn't effectively visible to the user (minimized, out of screen borders, etc).